instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyClientTracingListener.java

```diff
@@ -46,46 +46,48 @@ public final class JettyHttpClient9TracingInterceptor
     Request.QueuedListener.class
   };
 
-  @Nullable private Context context;
-
-  @Nullable
-  public Context getContext() {
-    return this.context;
-  }
-
-  private final Context parentContext;
-
+  private final Context context;
   private final Instrumenter<Request, Response> instrumenter;
 
-  public JettyHttpClient9TracingInterceptor(
-      Context parentCtx, Instrumenter<Request, Response> instrumenter) {
-    this.parentContext = parentCtx;
+  private JettyClientTracingListener(
+      Context context, Instrumenter<Request, Response> instrumenter) {
+    this.context = context;
     this.instrumenter = instrumenter;
   }
 
-  public void attachToRequest(Request jettyRequest) {
-    List<JettyHttpClient9TracingInterceptor> current =
-        jettyRequest.getRequestListeners(JettyHttpClient9TracingInterceptor.class);
-
+  @Nullable
+  public static Context handleRequest(
+      Context parentContext, HttpRequest request, Instrumenter<Request, Response> instrumenter) {
+    List<JettyClientTracingListener> current =
+        request.getRequestListeners(JettyClientTracingListener.class);
     if (!current.isEmpty()) {
-      logger.warning("A tracing interceptor is already in place for this request!");
-      return;
+      logger.warning("A tracing request listener is already in place for this request!");
+      return null;
+    }
+
+    if (!instrumenter.shouldStart(parentContext, request)) {
+      return null;
     }
-    startSpan(jettyRequest);
+
+    Context context = instrumenter.start(parentContext, request);
 
     // wrap all important request-based listeners that may already be attached, null should ensure
-    // are returned here
-    List<Request.RequestListener> existingListeners = jettyRequest.getRequestListeners(null);
-    wrapRequestListeners(existingListeners);
-
-    jettyRequest
-        .onRequestBegin(this)
-        .onRequestFailure(this)
-        .onResponseFailure(this)
-        .onResponseSuccess(this);
+    // that all listeners are returned here
+    List<Request.RequestListener> existingListeners = request.getRequestListeners(null);
+    wrapRequestListeners(existingListeners, context);
+
+    JettyClientTracingListener listener = new JettyClientTracingListener(context, instrumenter);
+    request
+        .onRequestBegin(listener)
+        .onRequestFailure(listener)
+        .onResponseFailure(listener)
+        .onResponseSuccess(listener);
+
+    return context;
   }
 
-  private void wrapRequestListeners(List<Request.RequestListener> requestListeners) {
+  private static void wrapRequestListeners(
+      List<Request.RequestListener> requestListeners, Context context) {
```

Comment:
looks like didn't end up using?
```suggestion
      List<Request.RequestListener> requestListeners) {
```

Reply:
It is used in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/227c7acc7520436f3c1f6a978e612c41b6a2cac5/instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyClientTracingListener.java#L115

Reply:
oh I see it was a field before, thanks